homeassistant/components/unmanic/sensor.py

```diff
@@ -0,0 +1,130 @@
+"""Sensor platform for Unmanic."""
+from homeassistant.components.sensor import SensorEntity
+
+from .const import DOMAIN, NAME
+from .entity import UnmanicEntity
+
+SENSORS = ["total-workers", "active-workers", "queue", "history"]
+
+
+async def async_setup_entry(hass, entry, async_add_devices):
+    """Unmanic setup sensor entity."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_devices([UnmanicSensor(coordinator, entry, key) for key in SENSORS])
+
+
+class UnmanicSensor(UnmanicEntity, SensorEntity):
+    """Unmanic Sensor class."""
+
+    def __init__(self, coordinator, config_entry, key) -> None:
+        """Initialize."""
+        self.coordinator = coordinator
+        self.config_entry = config_entry
+        self.key = key
+
+        super().__init__(
+            coordinator=coordinator,
+            config_entry=config_entry,
+            device_id=config_entry.entry_id,
+        )
+
+    @property
+    def unique_id(self):
+        """Return a unique ID to use for this entity."""
+        return f"{self.config_entry.entry_id}_{self.key}"
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        if self.key == "total-workers":
+            return f"{NAME} Total Workers"
+        elif self.key == "active-workers":
+            return f"{NAME} Active Workers"
+        elif self.key == "queue":
+            return f"{NAME} Queue"
+        elif self.key == "history":
+            return f"{NAME} History"
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if self.key == "total-workers":
+            return self.coordinator.data.get("settings").number_of_workers
+        elif self.key == "active-workers":
+            return len(self.coordinator.data.get("workers_status"))
+        elif self.key == "queue":
+            return self.coordinator.data.get("pending_tasks").recordsTotal
+        elif self.key == "history":
+            return self.coordinator.data.get("task_history").recordsTotal
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        if self.key == "total-workers":
+            return "workers"
+        elif self.key == "active-workers":
+            return "workers"
+        elif self.key == "queue":
+            return "files"
+        elif self.key == "history":
+            return "files"
+
+    @property
+    def extra_state_attributes(self):
+        """Return the state attributes."""
+        attrs = {}
```

Comment:
Why do you need all these extra state attributes? Shouldn't they just be extra entities?

Reply:
At any moment, there could be theoretically unlimited in the history or queue. This allows a quick look at the last 10 items in either the history or the queue. While they could be their own entities and I don't believe this is a great use of extra state attributes, I opted not to do that to not clutter up with 20 additional entities. What do you think is the best way I could represent this?